platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/VersionCatalogGenerator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins.internal;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.api.UncheckedIOException;
+import org.gradle.api.file.Directory;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+@NonNullApi
+public class VersionCatalogGenerator {
+    final VersionCatalogDependencyRegistry versionCatalogDependencyRegistry;
+    final Directory target;
+
+    public VersionCatalogGenerator(VersionCatalogDependencyRegistry versionCatalogDependencyRegistry, Directory target) {
+        this.versionCatalogDependencyRegistry = versionCatalogDependencyRegistry;
+        this.target = target;
+    }
+
+    public void generate() {
```

Comment:
🤔 I think it would be better if only the sections of the catalog with entries were actually generated.

Reply:
Changed.

Also generating the empty sections might get users intrigued to make even better use of version catalogs without them visiting the linked documentation? (like using bundles)